### PR TITLE
Use a Prometheus HA setup in the kube sample

### PR DESCRIPTION
**Pull request by alvaroaleman**

### CODE REVIEW

File: kube/manifests/prometheus-gcs.yaml [hunk 1]
```diff
@@ -18,16 +18,32 @@ spec:
...
-        - "--config.file=/etc/prometheus/prometheus.yml"
-        - "--storage.tsdb.path=/var/prometheus"
-        - "--storage.tsdb.min-block-duration=2h"
-        - "--storage.tsdb.max-block-duration=2h"
-        - "--web.enable-lifecycle"
+        command:
+        - "/bin/sh"
+        - "-c"
+        - >-
+            sed s/'@@HOSTNAME@@'/${HOSTNAME}/g /etc/prometheus/prometheus.yml.tmpl > /tmp/prometheus.yml
```
Comment (Bplotka) [inline kube/manifests/prometheus-gcs.yaml]: Yea, this one way doing it. 

But actually IMO more preferable would be to use thanos sidecar envvar substitution logic:

1. Use here ` --config.file=/etc/prometheus-shared/prometheus.yaml` as it is.
2. Move this volume mount to thanos sidecar
```
         - name: config
           mountPath: /etc/prometheus
``` 
3. Add shared dir for both containers:
```
          - mountPath: /etc/prometheus-shared
            name: shared-config
```
3.  Add these flags to sidecar:
```
 --reloader.config-file=/etc/prometheus/prometheus.yaml.tmpl
 --reloader.config-envsubst-file=/etc/prometheus-shared/prometheus.yaml
```
4.  Use `replica: $(HOSTNAME)` for prometheus.yaml.tmpl replica external label.

This is how we do for prod setup. 
The essential advantage of this is not using "2-process within one docker hack" we do here (:

What do you think?

**Comment by alvaroaleman** [comment]
Ah nice, I didn't see those flags before, definitely better than having to go through a shell ;)

This will work once #277 is in ;)

**Comment by Bplotka** [comment]
@alvaroaleman hm.. can you double check that? I think it works even without PR277:

* sidecar reloads in one go routine
* sidecar waits for prometheus in second go routine
* prometheus fails if config is not present and k8s restarts the container if that happens (if prom container starts before sidecar somehow)

Maybe I am wrong, but I think https://github.com/improbable-eng/thanos/pull/277 does not change anything. It does not matter in what order you put `funcs` into `run.Group`. There is no guarantee that go routine A when started before B, will actually run before B. Am I missing something?

**Comment by alvaroaleman** [comment]
For you this works without #277? Did you pull `improbable/thanos:latest`?

**Comment by Bplotka** [comment]
Ah, please use `improbable/thanos:master` instead for latest changes... Latest was meant for latest release, but there was none yet ): so it must be something old. Can update latest now for you as well.

**Comment by alvaroaleman** [comment]
Ahh, so that explains all the fuss :) I've updated this to use `:master` and will close #277 then

**Comment by Bplotka** [comment]
yea, sorry for the confusion. Updated `latest` just in case. When you will confirm it works, I will merge this.

**Comment by alvaroaleman** [comment]
Yes, with `:master` it worked all fine. I guess it makes more sense to leave it at `:master` since `:latest` will probably not get updated

**Comment by Bplotka** [comment]
Thanks!

**Pull request merged**